src/panels/lovelace/special-rows/hui-call-service-row.ts

```diff
@@ -1,25 +1,58 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement } from "@polymer/lit-element";
 import "@polymer/paper-button/paper-button.js";
 
 import "../../../components/ha-icon.js";
+
 import callService from "../common/call-service.js";
+import { EntityRow, CallServiceConfig } from "../entity-rows/types.js";
+import { HomeAssistant } from "../../../types.js";
+import { TemplateResult } from "lit-html";
+
+class HuiCallServiceRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: CallServiceConfig;
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public setConfig(config: CallServiceConfig): void {
+    if (
+      !config ||
+      !config.icon ||
+      !config.name ||
+      !config.action_name ||
+      !config.service ||
+      !config.service_data
```

Comment:
This is more of a question: Does Service Data need to be required? Isn't service data by default the entity_Id? Idk if that what is passed to this though

Reply:
The docs say it is required

Reply:
There is no entity_id in this config, so it can't default to entity ID. However, it's totally fine to have an empty service data. We should make it not required in a future PR.